platform/app/src/routes/Mode/Mode.tsx

```diff
@@ -25,14 +25,45 @@ const { getSplitParam } = utils;
  * @returns array of subscriptions to cancel
  */
 function defaultRouteInit(
-  { servicesManager, studyInstanceUIDs, dataSource, filters },
+  { servicesManager, studyInstanceUIDs, dataSource, filters, appConfig },
   hangingProtocolId
 ) {
   const {
     displaySetService,
     hangingProtocolService,
     uiNotificationService,
   } = servicesManager.services;
+  const minDisplaySetsToRunHP = appConfig?.minDisplaySetsToRunHP;
+
+  /**
+   * Function to apply the hanging protocol when the minimum number of display sets were
+   * received or all display sets retrieval were completed
+   * @returns
+   */
+  function applyHangingProtocol() {
+    // check if a hanging protocol is already applied. If so abort execution
+    if (hangingProtocolService.getActiveProtocol().protocol) {
```

Comment:
I still feel like this is not the correct check

Reply:
@wayfarer3130 Can you point the best suited test?

Reply:
This is all in one file so you can use a local variable to know whether or not you have applied a protocol in this mode.

Reply:
You don't want a check here at all - what you want is to just apply the hanging protocol when there first N items are applied, and the remaining items will simply continue retrieving until done.

Reply:
A created a variable inside the function that is set to true when the hanging protocol is applied

Reply:
The allSettle that calls the applyHangingProtocol is in study level. I dont have access to the series promises. And even if I had it, I wouldnt use because hangingprotocol sees displaySets, not seriesMetadata

Reply:
You do have access to the series promises - the allSettled below that I just commented on is where you have access to the series level promises.  You run the HP when enough series are loaded, not when the given display sets are actually available.